src/systems/app-mode.js

```diff
@@ -0,0 +1,227 @@
+/* global AFRAME, console, setTimeout, clearTimeout */
+
+const AppModes = Object.freeze({ DEFAULT: "default", HUD: "hud" });
+
+/**
+ * Simple system for keeping track of a modal app state
+ */
+AFRAME.registerSystem("app-mode", {
+  init() {
+    console.log("init app mode system");
+    this.setMode(AppModes.DEFAULT);
+  },
+
+  setMode(newMode) {
+    if (Object.values(AppModes).includes(newMode) && newMode !== this.mode) {
+      this.mode = newMode;
+      this.el.emit("appmode-change", { mode: this.mode });

```

Comment:
Maybe make the event name `"app-mode-change"` for consistency.

Reply:
✔️